wrappers/ios/tari_crypto.swift

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2021 The Tari Project
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the
+ * following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of
+ * its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Foundation
+
+enum Errors: Error {
+    case generic(_ errorCode: Int32)
+}
+
+class TariCrypto {
+
+    public func getVersion() -> String {
+        return String(cString: version())
+    }
+
+    public func lookupError(code: Int32) -> String {
+        // FFI interface needs to change, there is a safer and simpler way to do this
+        let length: Int32 = 129 // This can be eliminated, see below comment and conversion after that
+        var buffer: [Int8] = [Int8](repeating: 0, count: Int(length)) // Allocation/Deallocation should be kept on the same side of the boundary, only pointers and primitive types should cross, furthermore memory alignment is not guaranteed via FFI
```

Comment:
I think I should of used a better explanation here.

What I meant is the caller shouldn't be allocating/de-allocating memory or initializing variables for the library for any non-primitive type, that should stay on the library side, the caller merely requests in this case for it to be allocated and freed, then access to the value(s) is done via de-structuring of the type.

For this case, since most languages can handle cstrings without additional work, it colud look like the following:
```
int32 code = -1
cstringPtr = lookup_error_message(code)
let swiftstring = String(cString: cstringPtr)
free_cstring(cstringPtr)
return swiftstring
```

Reply:
Safest way to do it. Another possibility is to consider is `repr(C)`.